src/main/java/org/folio/search/configuration/OpensearchRestClientConfiguration.java

```diff
@@ -112,24 +128,59 @@ static class DefaultRestClientBuilderCustomizer implements RestClientBuilderCust
     @Override
     public void customize(HttpAsyncClientBuilder builder) {
       builder.setDefaultCredentialsProvider(new PropertiesCredentialsProvider(this.properties));
+      builder.setConnectionManager(getPoolingAsyncClientConnectionManager());
+      if (properties.isElasticsearchServer()) {
+        builder.addRequestInterceptorFirst((HttpRequest request, EntityDetails entityDetails, HttpContext context) -> {
+          var uri = request.getRequestUri();
+          if (uri.contains("cluster_manager_timeout")) {
+            var newUri = uri.replaceAll("[&?]?cluster_manager_timeout=[^&]*", "");
```

Comment:
Do you think it should work for all requests? Or just some specific one?
Should it be replaced with `master_timeout` query parameter instead of `cluster_manager_timeout`?